### PR TITLE
Update Swift binding

### PR DESCRIPTION
**Pull request by pelumi-ajayi**
# Description of change

Update swift binding with events

## Type of change

- Breaking change (feature that would cause existing functionality to not work as expected)

## How the change has been tested

Unit test in Xcode

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift [hunk 1]
```diff
@@ -16,12 +22,59 @@ class SwiftAwait: XCTestCase {
...
+    func testEvents() async throws {
+        let wallet = try! IotaWallet.Wallet()
+        let expectation = self.expectation(description: "TestEvents")
+
+        try! wallet.listen([]) { (message: String?, error: Error?) in
+            print("Event received: \(message!)")
+            expectation.fulfill()
+        }
+        
+        // TODO: Create account and do something that would trigger an event
```
Comment (Thoralf-M) [inline bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift]: Do you think it would be good to have a test event specially for such a case behind  a feature flag, so we don't have to do interactions that create an event, but can just call `emit_test_event()`?

Comment (pelumi-ajayi) [reply bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift]: Yeah, but I wouldn't know where such a function would fit.

The function would ideally be in wallet.rs, but calling it from here in Swift would mean there has to be a binding that's only used for testing, maybe such function should only exist behind a feature that's only available in debug mode to avoid misuse in production.

What do you think?

Comment (Thoralf-M) [reply bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift]: Is it possible to only be available in debug mode? Definitely behind a feature, but can't do any harm anyways

> Yeah, but I wouldn't know where such a function would fit.

I would just add it to the bottom of the AccountManager, the `verify_integrity()` method also only makes sense during development and should probably behind the same feature then

Comment (pelumi-ajayi) [reply bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift]: > Is it possible to only be available in debug mode? Definitely behind a feature, but can't do any harm anyways

I found this https://doc.rust-lang.org/reference/conditional-compilation.html#debug_assertions, but maybe it's fine putting it behind a feature.

> the verify_integrity() method also only makes sense during development and should probably behind the same feature then

Good point, I'll handle that as well.

Comment (pelumi-ajayi) [reply bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift]: Since Rust features don't translate to other languages, I eventually decided to go with making emit_test_event() available in debug builds only, I think it enforces the constraint that this should only be used while testing rather than in production.

File: src/account_manager/mod.rs [hunk 2]
```diff
@@ -165,13 +167,22 @@ impl AccountManager {
...
     /// automatically fix?) Also checks for each account if there is a gap in an address list and no address is
     /// duplicated
     pub async fn verify_integrity(&self) -> crate::Result<()> {
         verify_integrity(self).await
     }
 
+    #[cfg(debug_assertions)]
+    /// Checks if there is no missing account for example indexes [0, 1, 3] should panic (for now, later return error,
+    /// automatically fix?) Also checks for each account if there is a gap in an address list and no address is
+    /// duplicated
```
Comment (Thoralf-M) [inline src/account_manager/mod.rs]: Wrong comment
```suggestion
    /// Helper function to test events. Emits a provided event with account index 0.
```

Comment (pelumi-ajayi) [reply src/account_manager/mod.rs]: Thanks, don't know how I missed that.

File: bindings/swift/xcode/IotaWallet/IotaWallet.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -14,7 +14,8 @@
 		7904C23327BBDFB2001CB01B /* Wallet.m in Sources */ = {isa = PBXBuildFile; fileRef = 7904C23027BBDFB2001CB01B /* Wallet.m */; };
 		7904C23827BBDFD2001CB01B /* await.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7904C23627BBDFD2001CB01B /* await.swift */; };
 		7904C23927BBDFD2001CB01B /* callback.m in Sources */ = {isa = PBXBuildFile; fileRef = 7904C23727BBDFD2001CB01B /* callback.m */; };
-		7904C23C27BBE08D001CB01B /* libiota_wallet.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = 7904C23B27BBE08D001CB01B /* libiota_wallet.dylib */; };
+		79BD1BBB27CE54810043965C /* libiota_wallet.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = 79BD1BBA27CE54810043965C /* libiota_wallet.dylib */; };
```
Comment (Thoralf-M) [inline bindings/swift/xcode/IotaWallet/IotaWallet.xcodeproj/project.pbxproj:17]: Is this all auto generated and is this file needed? Or could we just gitignore it so we don't always get these changes?

**Review by Thoralf-M — APPROVED** [review]

File: bindings/swift/xcode/IotaWallet/IotaWallet.xcodeproj/project.pbxproj (hunk 3 above)
Comment (Thoralf-M) [reply bindings/swift/xcode/IotaWallet/IotaWallet.xcodeproj/project.pbxproj:17]: Feel free to merge if it's good to keep the file

Comment (pelumi-ajayi) [reply bindings/swift/xcode/IotaWallet/IotaWallet.xcodeproj/project.pbxproj:17]: This one is auto-generated based on Xcode settings I think, I'll add gitignore later on.
